### PR TITLE
[L2 switch mode] Update l2switch.j2 template

### PR DESCRIPTION
**Pull request by shi-su**
**- Why I did it**
The l2switch.j2 template does not include all fields for PORT. This could be incompatible with the 201911 image or later. 

**- How I did it**
Update l2switch.j2 template and add a unit test.

**- How to verify it**

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: src/sonic-config-engine/data/l2switch.j2 [hunk 1]
```diff
@@ -6,8 +6,11 @@
         {%- if ns.firstPrinted %},{% endif %}
 
         "{{ key }}": {
-            "alias": "{{ value.alias }}",
-            "lanes": "{{ value.lanes }}",
+            {%- for keyPort,valuePort in value.items() %}
```
Comment (qiluo-msft) [inline src/sonic-config-engine/data/l2switch.j2:9]: >keyPort [](start = 20, length = 7)

If keyPort == 'admin_status', is it a problem?

Comment (qiluo-msft) [inline src/sonic-config-engine/data/l2switch.j2:9]: >, [](start = 27, length = 1)

Add a blank after `,`. Also some more occurrences in the whole file.

**Comment by qiluo-msft** [comment]
>         argument = '-k Mellanox-SN2700 --preset l2 -p ' + self.t0_port_config

Could you add another test case for Mellanox-SN3800-D112C8 in the same way?

---
Refers to: src/sonic-config-engine/tests/test_j2files.py:99 in dd1f4d9. [](commit_id = dd1f4d96b213668c166d05a8ae3e1652dd0cb4f2, deletion_comment = False)

**Review by qiluo-msft — CHANGES_REQUESTED** [review]
As comments

File: src/sonic-config-engine/data/l2switch.j2 (hunk 1 above)
Comment (shi-su) [reply src/sonic-config-engine/data/l2switch.j2]: Added check to resolve the scenario with `admin_status` exists.

Comment (shi-su) [reply src/sonic-config-engine/data/l2switch.j2]: Fixed

**Comment by shi-su** [comment]
> > ```
> >     argument = '-k Mellanox-SN2700 --preset l2 -p ' + self.t0_port_config
> > ```
> 
> Could you add another test case for Mellanox-SN3800-D112C8 in the same way?
> 
> Refers to: src/sonic-config-engine/tests/test_j2files.py:99 in [dd1f4d9](https://github.com/Azure/sonic-buildimage/commit/dd1f4d96b213668c166d05a8ae3e1652dd0cb4f2). [](commit_id = [dd1f4d9](https://github.com/Azure/sonic-buildimage/commit/dd1f4d96b213668c166d05a8ae3e1652dd0cb4f2), deletion_comment = False)

I think I understand your point: different device types have different sources of port configuration data, and it would be better to capture the difference. However, in this particular test, the data source is explicitly specified (tests/t0-sample-port-config.ini). So only changing the device type cannot test out the sources of port configuration data, and in this test setup, it does not seem the source of port configuration data could follow the `-k <device>` option. In using the command in SONiC, the `-k <device>` option should specify the data, and rendering the template should not be dependent on what kind of device it is.

**Review by qiluo-msft — APPROVED** [review]

**Comment by shi-su** [comment]
retest vsimage please

**Pull request merged**